File: Algorithms/Java/MatrixMultiplication.java
public class MatrixMultiplication {

    public static void main(String[] args) {

        int[][] matrix1 = { { 1, 2, 3 }, { 4, 5, 6 } };
        int[][] matrix2 = { { 7, 8 }, { 9, 10 }, { 11, 12 } };

        int rows1 = matrix1.length;
        int cols1 = matrix1[0].length;
        int cols2 = matrix2[0].length;

        if (cols1 != matrix2.length) {
            System.out.println("Matrices cannot be multiplied.");
            return;
        }

        int[][] result = new int[rows1][cols2];

        for (int i = 0; i < rows1; i++) {
            for (int j = 0; j < cols2; j++) {
                for (int k = 0; k < cols1; k++) {
                    result[i][j] += matrix1[i][k] * matrix2[k][j];
                }
            }
        }

        System.out.println("Result matrix: ");
        for (int i = 0; i < rows1; i++) {
            for (int j = 0; j < cols2; j++) {
                System.out.print(result[i][j] + " ");
            }
            System.out.println();
        }
    }
}

File: Algorithms/Javascript/matrixMultiplication.js
const multiplyMatrix = (matrix1, matrix2) => {
    const matrixSize = matrix1.length;
    const resultarr = new Array(matrixSize);

    for (let i = 0; i < matrixSize; i++) {
        resultarr[i] = new Array(matrixSize);
    }

    for (let i = 0; i < matrixSize; i++) {
        for (let j = 0; j < matrixSize; j++) {
            resultarr[i][j] = 0;
            for (let k = 0; k < matrixSize; k++) {
                resultarr[i][j] += matrix1[i][k] * matrix2[k][j];
            }
        }
    }

    return resultarr;
};

const A = [[1, 2], [3, 4]];
const B = [[1, 2], [3, 4]];
const C = multiplyMatrix(A, B);

console.log(C);
